src/helm/benchmark/contamination/compute_contamination_metrics.py

```diff
@@ -0,0 +1,248 @@
+import json
+import argparse
+import os
+import glob
+
+from typing import List, Tuple, Set
+from nltk import ngrams
+from typing import Dict, Optional, DefaultDict
+from collections import defaultdict
+
+from helm.benchmark.contamination.light_scenario import LightInstance, LightScenario
+from helm.benchmark.contamination.light_tokenizer import LightTokenizer, DefaultTokenizer
+from helm.benchmark.contamination.document_reading_processor import DocumentReadingProcessor
+from helm.benchmark.contamination.contamination_stats import ContaminationStats, PART_INPUT, PART_REF
+
+
+# The n values of the ngrams to be computed
+N_VALUES: List[int] = [5, 9, 13]  # TODO: Pick the N values
+
+
+def load_scenarios_from_jsonl(path: str) -> List[LightScenario]:
+    """
+    Create a list of light scenarios from a jsonl file, where each json represents a LightScenario object.
+
+    Input file format:
+
+    Instance JSON 1
+    Instance JSON 2
+    Instance JSON 3
+    ...
+
+    Each line is a json and each json looks like:
+    {
+        "scenario_spec": {
+            "name": "SCENARIO_NAME",
+            "split": "SPLIT"
+        },
+        "light_instances": [
+            {
+                "input": "INPUT_TEXT1",
+                "references": [
+                    "REFERENCE_TEXT_1",
+                    "REFERENCE_TEXT_2"
+                ]
+            },
+            {
+                "input": "INPUT_TEXT2",
+                "references": [
+                    "REFERENCE_TEXT_3",
+                    "REFERENCE_TEXT_4"
+                ]
+            }
+        ]
+    }
+    """
+
+    def create_light_instance_from_dict(instance_dict: dict) -> LightInstance:
+        return LightInstance(input=instance_dict["input"], references=instance_dict["references"])
+
+    light_scenarios: List[LightScenario] = []
+    scenario_jsons = open(path, "r").readlines()
+    for scenario_json in scenario_jsons:
+        scenario_dict: dict = json.loads(scenario_json)
+        scenario_spec: dict = scenario_dict["scenario_spec"]
+        light_instances: List[LightInstance] = [
+            create_light_instance_from_dict(instance_dict) for instance_dict in scenario_dict["light_instances"]
+        ]
+        light_scenarios.append(LightScenario(scenario_spec=scenario_spec, light_instances=light_instances))
+    return light_scenarios
+
+
+def create_stats_and_ngram_index(
+    scenarios: List[LightScenario], n_values: List[int], tokenizer: LightTokenizer
+) -> Tuple[Dict[str, ContaminationStats], DefaultDict[int, DefaultDict[Tuple[str], Set[tuple]]]]:
```

Comment:
Make a data class to capture the return type...complex types are hard to read and reasonable about the semantics.